scripts/report_incomplete_assignments_with_work_orders.py

```diff
@@ -0,0 +1,151 @@
+# -*- coding: UTF-8 -*-
+"""
+   Copyright 2020 Esri
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+
+   you may not use this file except in compliance with the License.
+
+   You may obtain a copy of the License at
+   
+   http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+
+   distributed under the License is distributed on an "AS IS" BASIS,
+
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+
+   See the License for the specific language governing permissions and
+
+   limitations under the License.​
+
+   This sample logs incomplete assignments without a corresponding survey / collector feature
+"""
+
+import argparse
+import logging
+import logging.handlers
+import sys
+import traceback
+from arcgis.apps import workforce
+from arcgis.gis import GIS
+from arcgis.features import FeatureLayer
+
+
+def initialize_logging(log_file=None):
+	"""
+	Setup logging
+	:param log_file: (string) The file to log to
+	:return: (Logger) a logging instance
+	"""
+	# initialize logging
+	formatter = logging.Formatter(
+		"[%(asctime)s] [%(filename)30s:%(lineno)4s - %(funcName)30s()][%(threadName)5s] [%(name)10.10s] [%(levelname)8s] %(message)s")
+	# Grab the root logger
+	logger = logging.getLogger()
+	# Set the root logger logging level (DEBUG, INFO, WARNING, ERROR, CRITICAL)
+	logger.setLevel(logging.DEBUG)
+	# Create a handler to print to the console
+	sh = logging.StreamHandler(sys.stdout)
+	sh.setFormatter(formatter)
+	sh.setLevel(logging.INFO)
+	# Create a handler to log to the specified file
+	if log_file:
+		rh = logging.handlers.RotatingFileHandler(log_file, mode='a', maxBytes=10485760)
+		rh.setFormatter(formatter)
+		rh.setLevel(logging.DEBUG)
+		logger.addHandler(rh)
+	# Add the handlers to the root logger
+	logger.addHandler(sh)
+	return logger
+
+
+def main(arguments):
+	# Initialize logging
+	logger = initialize_logging(arguments.log_file)
+
+	# Create the GIS
+	logger.info("Authenticating...")
+
+	# First step is to get authenticate and get a valid token
+	gis = GIS(arguments.org_url,
+			  username=arguments.username,
+			  password=arguments.password,
+			  verify_cert=not arguments.skip_ssl_verification)
+
+	logger.info("Getting workforce project")
+
+	# Get the workforce project
+	item = gis.content.get(arguments.project_id)
+	try:
+		project = workforce.Project(item)
+	except Exception:
+		logger.info("Invalid project id")
+		sys.exit(0)
+		
+	layer = None
+	# Get Survey or Collector Feature Layer
+	if arguments.survey_id and arguments.layer_url:
+		logger.info("Please try again with either survey id or layer url provided, not both")
+		sys.exit(0)
+	elif arguments.survey_id:
+		survey_item = gis.content.get(arguments.survey_id)
+		if survey_item:
+			layer = survey_item.layers[0]
```

Comment:
maybe log that survey is invalid if it's `None`?

Reply:
This is handled later on in the script, as the `layer` object is set to None